### PR TITLE
Iris Rev 4 - modifier thumb cluster and dual encoder support

### PR DESCRIPTION
**Pull request by compilation-error**
## Description
Keebio Iris rev4 layout. The modifiers are set up in the thumb clusters. Supports dual encoders - one for volume and one for scrolling. Mouse support is enabled on layer 1 along with keypad. Layer 2 is mostly default with reset and EEPROM reset moved to thumb cluster.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/iris/keymaps/compilation-error/keymap.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/compilation-error/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/keebio/iris/keymaps/compilation-error/keymap.c [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 3
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/compilation-error/keymap.c:7]:
```suggestion
enum custom_layers {
    _QWERTY,
    _LOWER,
    _RAISE,
    _ADJUST,
};
```

File: keyboards/keebio/iris/keymaps/compilation-error/keymap.c [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    }
+    else if (index == 1) {
+        if (clockwise) {
+            tap_code(KC_WH_D);
+        } else {
+            tap_code(KC_WH_U);
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/compilation-error/keymap.c:130]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
    if (index == 0) {
        if (clockwise) {
            tap_code(KC_VOLU);
        } else {
            tap_code(KC_VOLD);
        }
    }
    else if (index == 1) {
        if (clockwise) {
            tap_code(KC_WH_D);
        } else {
            tap_code(KC_WH_U);
        }
    }
    return true;
}
```

File: keyboards/keebio/iris/keymaps/compilation-error/rules.mk [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/compilation-error/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by compilation-error** [comment]
thank you for your review and suggested updates. I will incorporate them and create a new PR - this layout has some deficiencies.

**Pull request closed without merging**